float64 latitude
float64 longitude
float64 heading
bool gotGps
bool gotHeading
